### ISSUE TITLE
Live reloading with custom JS

### ISSUE THREAD

**Issue by mattvr** [cezaraugusto/extension.js#62]
I have a custom JS script I'm separately building and including in my extension (outside of webpack).

If I include it in `./scripts/`, I get a `$RefreshSig$ is not defined` error as I believe the webpack processing on top is causing a conflict.

If I include it in `./public/`, everything works great, but live reloading doesn't work and I have to restart the browser fully every time my script changes.

Ideally, I'd like for my custom JS (and other assets like images) to refresh without restarting the browser or throwing the webpack error.

No worries if out of scope, just figured I'd ask if there's an existing option or a bug I'm hitting – I could feasibly work around this.

Thank you for the awesome tool!

**Comment by cezaraugusto** [comment]
Hi there! Thanks!

If you are trying to load a source (not bundled) file in `scripts/` and you get a `$RefreshSig$ is not defined`, that's a bug. 

However if this file is bundled somewhere else outside `Extension.js` (like you said -- separately building and including in your extension), then `scripts/` will fail if there are variables and imports outside the scope, as it has no access to the external bundling system. In this case, adding them to `public/` might be the right choice. However, since it was designed for static assets, it does not listen for file changes.

I think having `public/` to listen to file changes wouldn't affect existing functionality, but this is currently far from priorities. If you're feeling adventurous, editing the [CopyPublicFolder](https://github.com/cezaraugusto/extension.js/blob/8f81fe2057516e74a539351f840cdd156f316104/programs/develop/webpack/plugins/SpecialFoldersPlugin/CopyPublicFolder.ts#L70-L73) file is the alternative for handling reload of `public/` assets. I theorize that adding a listener to the `change` event in there would make your request possible. Something like:

```ts
watcher.on('change', (filePath: string) => {
  const target = path.join(output, path.relative(projectPath, filePath))
  this.copyFile(filePath, target)
})
```

The manual test would be adding a file to `public/` and asserting that Extension.js reloads if they change. I added some instructions to run the project locally in [CONTRIBUTING.md](https://github.com/cezaraugusto/extension.js/blob/main/CONTRIBUTING.md). If you want to give it a go, I can definitely review.

**Comment by mattvr** [comment]
Great tip, that did it! @cezaraugusto I've tested and submitted a PR for here and updated docs, linked above.

**Comment by cezaraugusto** [comment]
Fixed by #64
